asset/asset.go

```diff
@@ -756,47 +743,61 @@ func NewRawKeyGenesisSigner(priv *btcec.PrivateKey) *RawKeyGenesisSigner {
 // For minting the first asset in a group, only one Genesis object is
 // needed, since we tweak with and sign over the same Genesis object.
 // The final tweaked public key and the signature are returned.
-func (r *RawKeyGenesisSigner) SignGenesis(keyDesc keychain.KeyDescriptor,
-	initialGen Genesis, currentGen *Genesis) (*btcec.PublicKey,
-	*schnorr.Signature, error) {
+func (r *RawKeyGenesisSigner) SignVirtualTx(signDesc *lndclient.SignDescriptor,
+	virtualTx *wire.MsgTx, prevOut *wire.TxOut) (*schnorr.Signature,
+	error) {
+
+	signerPubKey := r.privKey.PubKey()
 
-	if !keyDesc.PubKey.IsEqual(r.privKey.PubKey()) {
-		return nil, nil, fmt.Errorf("cannot sign with key")
+	if !signDesc.KeyDesc.PubKey.IsEqual(signerPubKey) {
+		return nil, fmt.Errorf("cannot sign with key")
 	}
 
-	// TODO(jhb): Update to two-phase tweak
-	tweakedPrivKey := txscript.TweakTaprootPrivKey(
-		*r.privKey, initialGen.GroupKeyTweak(),
-	)
+	sig, err := SignVirtualTx(r.privKey, signDesc, virtualTx, prevOut)
+	if err != nil {
+		return nil, err
+	}
 
-	// If the current genesis is not set, we are minting the first asset in
-	// the group. This means that we use the same Genesis object for both
-	// the key tweak and to create the asset ID we sign. If the current
-	// genesis is set, the asset type of the new asset must match the type
-	// of the first asset in the group.
-	id := initialGen.ID()
-	if currentGen != nil {
-		if initialGen.Type != currentGen.Type {
-			return nil, nil, fmt.Errorf("asset group type mismatch")
-		}
+	return sig, nil
+}
+
+// A compile-time assertion to ensure RawKeyGenesisSigner meets the
+// GenesisSigner interface.
+var _ GenesisSigner = (*RawKeyGenesisSigner)(nil)
+
+type RawGroupTxBuilder struct{}
 
-		id = currentGen.ID()
+func (m *RawGroupTxBuilder) BuildGenesisTx(newAsset *Asset) (*wire.MsgTx,
```

Comment:
Does this version need to live in the `asset` package? Can we get away with just the version in `tapscript`?

Reply:
Not really, since that would cause dependency cycles for many tests that use `RandGroupKey()`. Similar to how we had the raw key genesis signer and the real `gen_signer`.